### PR TITLE
Whatsapp

### PR DESCRIPTION
**Pull request by xprmnts**
1) Added values of the browser viewport to the ShareButtons state and rendered bootstrap columns and whatsapp icons based on viewport values.

2) Modified css to display whatsapp icon only for devices under 600 px width.

### CODE REVIEW

**Comment by logophobia** [comment]
Which CA item is this?

**Comment by xprmnts** [comment]
Oops, it is CA -376

File: CollAction/Frontend/app/global/ShareButtons.tsx [hunk 1]
```diff
@@ -14,98 +14,138 @@ interface IShareButtonsState {
...
-                <i className="fa fa-twitter"></i>
-              </div>
-            </a>
-          </div>
-          <div className="col-xs-3">
-            <a href={this.getLinkedInUrl()}>
-              <div className="social-media-share-buttons social-media-share-button-linkedin">
-                <i className="fa fa-linkedin"></i>
+      return (
+          <div className="share-buttons">
```
Comment (ljwboschker) [inline CollAction/Frontend/app/global/ShareButtons.tsx]: It's probably better to lay out the social media buttons in one bootstrap column, with a little margin. That way, you don't need the javascript-code to listen to the window size and determine wether you need xs-2 or xs-3.

Like this:

      <div className="share-buttons">
        <div className="row">
          <div className="col-xs-3 share-count">
            {this.state.shareCount}<br /> Shares
          </div>
          <div className="col-xs-9">
            <a href={this.getFacebookUrl()} target="_blank">
              <div className="social-media-share-buttons social-media-share-button-facebook">
                <i className="fa fa-facebook"></i>
              </div>
            </a>
            <a href={this.getTwitterUrl()}>
              <div className="social-media-share-buttons social-media-share-button-twitter">
                <i className="fa fa-twitter"></i>
              </div>
            </a>
            <a href={this.getLinkedInUrl()}>
              <div className="social-media-share-buttons social-media-share-button-linkedin">
                <i className="fa fa-linkedin"></i>
              </div>
            </a>
            <a href={this.getLinkedInUrl()}>
              <div className="social-media-share-buttons social-media-share-button-whatsapp">
                <i className="fa fa-whatsapp"></i>
              </div>
            </a>
          </div>
        </div>
      </div>

Then add this to the .social-media-share-buttons class:

    float: left;
    margin-right: 5px;

Comment (ljwboschker) [reply CollAction/Frontend/app/global/ShareButtons.tsx]: Perhaps you can find a way to remove the margin-right: 5px for the last button; I had some trouble figuring that out.

File: CollAction/Frontend/app/global/ShareButtons.tsx [hunk 2]
```diff
@@ -14,98 +14,138 @@ interface IShareButtonsState {
   shareCount?: number;
   shareCountError?: Error;
   shareCountState: States;
+    width: number;
```
Comment (logophobia) [inline CollAction/Frontend/app/global/ShareButtons.tsx]: The indentation of the whole file and changeset seems a bit inconsistent. Make sure you try to emulate the indentation style of the other typescript files.

File: CollAction/Frontend/app/global/ShareButtons.tsx [hunk 3]
```diff
@@ -125,6 +165,9 @@ const FullShareButtons = () => {
   );
 };
 
+
```
Comment (logophobia) [inline CollAction/Frontend/app/global/ShareButtons.tsx]: Unnecessary blank lines?

File: CollAction/Properties/launchSettings.json [hunk 4]
```diff
@@ -4,14 +4,14 @@
     "anonymousAuthentication": true,
     "iisExpress": {
       "applicationUrl": "https://localhost:44301/",
-      "sslPort": 44301
```
Comment (logophobia) [inline CollAction/Properties/launchSettings.json]: Please don't check these changes in, disables ssl for local testing I think.

**Pull request closed without merging**